File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Solution {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		for (int t = 1; t <= 10; t++) {
			int count = 0;
			int N = Integer.parseInt(br.readLine());
			int[] arr = new int[N];
			// 건물들 배열로 입력 받기
			// test 입력 => 0 0 3 5 2 4 9 0 6 4 0 6 0 0 (14개)
			StringTokenizer st = new StringTokenizer(br.readLine());
			for (int i = 0; i < arr.length; i++) {
				arr[i] = Integer.parseInt(st.nextToken());
			}
			// 처음,끝 인덱스 값 두 개는 0으로 고정 -> 2 ~ arr.length - 2 까지
			for (int i = 2; i < arr.length - 2; i++) {
				int leftmax = Math.max(arr[i - 2], arr[i - 1]);
				int rightmax = Math.max(arr[i + 1], arr[i + 2]);

				if (arr[i] > leftmax && arr[i] > rightmax) {
					int cnt = Math.min(arr[i] - leftmax, arr[i] - rightmax);
					count += cnt;
				}
			}
			System.out.println("#" + t + " " + count);
		}
	}
}

File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.py
for test_case in range(1, 11):
    n = int(input())
    arr = list(map(int, input().split()))
    count = 0
    
    for node in range(2, n-2):
        # 가장 큰 값 찾기
        if arr[node] > arr[node-2] and arr[node] > arr[node-1] and arr[node] > arr[node+1] and arr[node] > arr[node+2]:
            max_building = arr[node]

            # 왼쪽 높은 값
            left_max = max(arr[node-2], arr[node-1])
            # 오른쪽 높은 값
            right_max = max(arr[node+1], arr[node+2])

            side_max = max(left_max, right_max)
            count += max_building - side_max

    print("#{} {}" .format(test_case, count))